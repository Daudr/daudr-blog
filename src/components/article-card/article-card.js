import React from 'react'
import { Link } from 'gatsby'

import { makeStyles } from '@material-ui/styles'
import Card from '@material-ui/core/Card'
import CardMedia from '@material-ui/core/CardMedia'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'
import Chip from '@material-ui/core/Chip'
import { rhythm } from '../../utils/typography'
import kebabCase from 'lodash/kebabCase'

const useStyles = makeStyles({
	card: {
		minWidth: 275,
		marginBottom: rhythm(1 / 2),
	},
	title: {
		fontSize: 14,
	},
	pos: {
		marginBottom: 12,
	},
	chip: {
		margin: `2px`,
	},
	media: {
		height: 280,
	},
})

export const ArticleCard = ({ node, isIndex = false, isIT = false }) => {
	const title = node.frontmatter.title || node.fields.slug
	const tags = node.frontmatter.tags || []
	const classes = useStyles()

	return (
		<Link to={node.fields.slug} className='article-card__link'>
			<Card className={classes.card}>
				<CardMedia
					className={classes.media}
					image={node.frontmatter.cover_image}
					title={title}
				/>
				<CardContent>
					<Typography
						className={classes.title}
						color='textSecondary'
						gutterBottom
					>
						{tags.slice(0, 3).map(tag => {
							return (
								<Link
									key={tag}
									to={`/tags/${isIT ? `it/` : ``}${kebabCase(tag)}`}
									style={{ boxShadow: `none` }}
								>
									<Chip
										size='small'
										label={`#${tag}`}
										className={classes.chip}
										style={{
											fontFamily: `'Anton', sans-serif`,
											fontWeight: `bold`,
											textTransform: `uppercase`,
										}}
										onClick={() => {}}
									></Chip>
								</Link>
							)
						})}
					</Typography>
					<Typography variant='h5' component='h2' data-cy='post-title'>
						{title}
					</Typography>
					<Typography
						className={classes.pos}
						variant='subtitle2'
						component='p'
						color='textSecondary'
					>
						{node.frontmatter.date}
					</Typography>

					<Typography variant='body2' component='p'>
						{node.frontmatter.description || node.excerpt}
					</Typography>
				</CardContent>
			</Card>
		</Link>
	)
}

export default ArticleCard
