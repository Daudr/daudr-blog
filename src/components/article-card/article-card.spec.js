import React from "react"
import renderer from "react-test-renderer"

import ArticleCard from "./article-card"

import { node } from "../../../__mocks__/article-card.mock"

describe("ArticleCard", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ArticleCard node={node} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
