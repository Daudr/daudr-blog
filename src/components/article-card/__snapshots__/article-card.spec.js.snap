// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard renders correctly 1`] = `
<a
  href="/ngfor-done-right/"
  style={
    Object {
      "boxShadow": "none",
    }
  }
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardMedia-root makeStyles-media-5"
      style={
        Object {
          "backgroundImage": "url(\\"https://images.unsplash.com/photo-1489976908522-aabacf277f49?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80\\")",
        }
      }
      title="ngFor Done Right"
    />
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root makeStyles-title-2 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
      >
        <a
          href="/tags/angular"
          style={
            Object {
              "boxShadow": "none",
            }
          }
        >
          <div
            className="MuiButtonBase-root MuiChip-root makeStyles-chip-4 MuiChip-sizeSmall MuiChip-clickable"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            style={
              Object {
                "fontFamily": "'Anton', sans-serif",
                "fontWeight": "bold",
                "textTransform": "uppercase",
              }
            }
            tabIndex={0}
          >
            <span
              className="MuiChip-label MuiChip-labelSmall"
            >
              #angular
            </span>
          </div>
        </a>
        <a
          href="/tags/code"
          style={
            Object {
              "boxShadow": "none",
            }
          }
        >
          <div
            className="MuiButtonBase-root MuiChip-root makeStyles-chip-4 MuiChip-sizeSmall MuiChip-clickable"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            style={
              Object {
                "fontFamily": "'Anton', sans-serif",
                "fontWeight": "bold",
                "textTransform": "uppercase",
              }
            }
            tabIndex={0}
          >
            <span
              className="MuiChip-label MuiChip-labelSmall"
            >
              #code
            </span>
          </div>
        </a>
        <a
          href="/tags/guide"
          style={
            Object {
              "boxShadow": "none",
            }
          }
        >
          <div
            className="MuiButtonBase-root MuiChip-root makeStyles-chip-4 MuiChip-sizeSmall MuiChip-clickable"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            style={
              Object {
                "fontFamily": "'Anton', sans-serif",
                "fontWeight": "bold",
                "textTransform": "uppercase",
              }
            }
            tabIndex={0}
          >
            <span
              className="MuiChip-label MuiChip-labelSmall"
            >
              #guide
            </span>
          </div>
        </a>
      </p>
      <h2
        className="MuiTypography-root MuiTypography-h5"
        data-cy="post-title"
      >
        ngFor Done Right
      </h2>
      <p
        className="MuiTypography-root makeStyles-pos-3 MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
      >
        July 04, 2019
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2"
      >
        When dealing with lists in templates, ngFor saves us a lot of time, but it could be a real pain if used wrong, read this article to know hot use it right
      </p>
    </div>
  </div>
</a>
`;
